File: go/002/002.go
package main

import "fmt"

func main() {
	fibs := []uint64{0, 1}

	for fibs[len(fibs)-1]+fibs[len(fibs)-2] < 4000000 {
		fibs = append(fibs, fibs[len(fibs)-1]+fibs[len(fibs)-2])
	}

	var sum uint64 = 0

	for i := 0; i < len(fibs); i++ {
		if fibs[i]%2 == 0 {
			sum += fibs[i]
		}
	}

	fmt.Println(sum)
}

File: rust/002/002.rs
fn main() {
    let mut i = 0;
    let mut j = 1;
    let mut k = i + j;
    let mut sum = 0;

    while k < 4000000 {
        if k % 2 == 0 {
            sum += k;
        }

        i = j;

        j = k;

        k = i + j;
    }

    println!("{}", sum);
}
